drivers/rtc/rtc_ll_stm32.c

```diff
@@ -68,14 +70,22 @@ struct rtc_stm32_data {
 	struct k_mutex lock;
 };
 
-static int rtc_stm32_enter_initialization_mode(void)
+static int rtc_stm32_enter_initialization_mode(bool no_kernel)
```

Comment:
`no_kernel` as a bool is confusing. Please use `kernel_available` or similar that would avoid confusion